src/game.cpp

```diff
@@ -8213,12 +8215,84 @@ tripoint game::look_debug()
 
 void game::print_all_tile_info( const tripoint &lp, WINDOW *w_look, int column, int &line,
-                                bool mouse_hover )
+                                const int last_line, visibility_type visibility )
+{
+    switch( visibility ) {
+        case VIS_CLEAR:
+            print_terrain_info( lp, w_look, column, line );
+            print_fields_info( lp, w_look, column, line );
+            print_trap_info( lp, w_look, column, line );
+            print_object_info( lp, w_look, column, line, last_line );
+            break;
+        case VIS_BOOMER:
+        case VIS_BOOMER_DARK:
+        case VIS_DARK:
+        case VIS_LIT:
+        case VIS_HIDDEN:
+            print_visibility_info( w_look, column, line, visibility );
+            break;
+        default:
+            dbg(D_ERROR) << "game::print_all_tile_info: Unexpected visibility_type " << visibility;
+            debugmsg("game::print_all_tile_info: Unexpected visibility_type %d", visibility);
+            return;
+    }
+
+    if( m.has_graffiti_at( lp ) ) {
+        mvwprintw(w_look, ++line + 1, 1, _("Graffiti: %s"), m.graffiti_at( lp ).c_str() );
+    }
+
+    auto this_sound = sounds::sound_at( lp );
+    if( !this_sound.empty() ) {
+        mvwprintw( w_look, ++line, 1, _("You heard %s from here."), this_sound.c_str() );
+    } else {
+        // Check other z-levels
+        tripoint tmp = lp;
+        for( tmp.z = -OVERMAP_DEPTH; tmp.z <= OVERMAP_HEIGHT; tmp.z++ ) {
+            if( tmp.z == lp.z ) {
+                continue;
+            }
+
+            auto zlev_sound = sounds::sound_at( tmp );
+            if( !zlev_sound.empty() ) {
+                mvwprintw( w_look, ++line, 1, tmp.z > lp.z ?
+                           _("You heard %s from above.") : _("You heard %s from below."),
+                           zlev_sound.c_str() );
+            }
+        }
+    }
+}
+
+void game::print_visibility_info( WINDOW *w_look, int column, int &line,
+                                  visibility_type visibility )
 {
-    print_terrain_info( lp, w_look, column, line );
-    print_fields_info( lp, w_look, column, line );
-    print_trap_info( lp, w_look, column, line );
-    print_object_info( lp, w_look, column, line, mouse_hover );
+    const char* visibility_message = nullptr;
+    switch( visibility ) {
+        case VIS_CLEAR:
+            visibility_message = _("Clearly visible.");
+            break;
+        case VIS_BOOMER:
+            visibility_message = _("A bright pink blur.");
+            break;
+        case VIS_BOOMER_DARK:
+            visibility_message = _("A pink blur.");
+            break;
+        case VIS_DARK:
+            visibility_message = _("Darkness.");
+            break;
+        case VIS_LIT:
+            visibility_message = _("Bright light.");
+            break;
+        case VIS_HIDDEN:
+            visibility_message = _("Unseen.");
+            break;
+        default:
```

Comment:
In the updated commits, removed those error messages in the default cases in favor of compiler warnings.